Makefile

```diff
@@ -136,6 +137,24 @@ $(TEST): $(SOURCE) $(SPECS) $(BIN)
 testem:
 	$(BIN)/testem
 
+# Rules to attach checksums to JavaScript source in the header
+app.html: app.html.in .make-script-block
+	sed -e '/%SCRIPTS%/ {' -e 'r .make-script-block' -e 'd' -e '}' app.html.in > app.html
+
+.make-script-block: $(SOURCE)
+	> .make-script-block
+	for file in $(SOURCE); do \
+		checksum=$$($(MD5) $$file | awk '{ print $$1 }') ; \
+		echo "<script src='$$file?v=$$checksum'></script>" >> .make-script-block ; \
+	done
+
+# Rule to determine MD5 utility
```

Comment:
Did you mean I should move the MD5 conditional to the top? (It wouldn't make sense to put the app.html rules to the top right?)